web/src/components/Breadcrumbs.tsx

```diff
@@ -45,42 +51,92 @@ const StyledIcon = styled(Icon)`
 `
 
 type BreadcrumbsProps = {
   ancestorBreadcrumbs: BreadcrumbModel[]
   currentBreadcrumb: BreadcrumbModel
 }
 
+const getLastBreadcrumbs = (
+  ancestorBreadcrumbs: BreadcrumbModel[],
+  currentBreadcrumb: BreadcrumbModel,
+): BreadcrumbModel[] => {
+  const allBreadcrumbs = [...ancestorBreadcrumbs, currentBreadcrumb]
+  const breadCrumbsLimit = 3 // with home included
+  const lastTwoCrumbs = -2
+
+  if (ancestorBreadcrumbs.length === 0) {
+    return []
+  }
+
+  if (allBreadcrumbs.length <= breadCrumbsLimit) {
+    return allBreadcrumbs
+  }
+
+  const home = allBreadcrumbs[0] as BreadcrumbModel
+  const rest = allBreadcrumbs.slice(1)
+  const ellipsis = new BreadcrumbModel({
+    title: '...',
+    pathname: '',
+    node: <StyledEllipsis>...</StyledEllipsis>,
+  })
+
+  return [home, ellipsis, ...rest.slice(lastTwoCrumbs)]
+}
+
 const Breadcrumbs = ({ ancestorBreadcrumbs, currentBreadcrumb }: BreadcrumbsProps): ReactElement => {
   // The current page should not be listed in the UI, but should be within the JsonLd.
   const jsonLdBreadcrumbs = [...ancestorBreadcrumbs, currentBreadcrumb]
   // Min text length after which the last breadcrumb item should shrink
   const MIN_SHRINK_CHARS = 20
 
-  /* We do some funky stuff with directions here. See this link for more information about the idea:
-   https://css-tricks.com/position-vertical-scrollbars-on-opposite-side-with-css/
-   Basically, we are inverting the direction on the wrapper and then making sure that the direction of the content
-   has the opposite direction of the wrapper. */
   return (
-    <Wrapper>
+    <StyledBox>
       <JsonLdBreadcrumbs breadcrumbs={jsonLdBreadcrumbs} />
-      <OrderedList>
-        {ancestorBreadcrumbs.map((breadcrumb, index) =>
-          ancestorBreadcrumbs.length > 1 && index === 0 ? (
-            <li key={breadcrumb.pathname}>
-              <StyledLink to={breadcrumb.pathname}>
+      <StyledMuiBreadcrumbs length={ancestorBreadcrumbs.length} aria-label='breadcrumb' separator={<Separator />}>
+        {getLastBreadcrumbs(ancestorBreadcrumbs, currentBreadcrumb).map((breadcrumb, index, array) => {
+          const isHome = array.length > 1 && index === 0
+          const isLast = index === array.length - 1
+
+          if (isHome) {
+            return (
+              <StyledLink key={breadcrumb.pathname} to={breadcrumb.pathname}>
                 <StyledIcon src={HomeOutlinedIcon} title={breadcrumb.title} />
               </StyledLink>
-            </li>
-          ) : (
-            <Breadcrumb key={breadcrumb.title} shrink={breadcrumb.title.length >= MIN_SHRINK_CHARS}>
-              {breadcrumb.node}
-            </Breadcrumb>
-          ),
-        )}
-      </OrderedList>
-    </Wrapper>
+            )
+          }
+
+          if (breadcrumb.title === '...') {
+            return <StyledEllipsis key='ellipsis'>...</StyledEllipsis>
+          }
+
+          return (
+            <Breadcrumb
+              title={breadcrumb.title}
```

Comment:
what if we use collapsed breadcrumbs to avoid slicing and shrinking them like it is shown [here](https://mui.com/material-ui/react-breadcrumbs/#collapsed-breadcrumbs) :thinking:

Reply:
Hmm this is a good idea but still if the user unfold it it will take the same same space...
Unless we want to enable that for just for desktop.

Reply:
you are right here :thinking: